File: analyst/src/main/java/com/zorroa/analyst/service/ShutdownHook.java
package com.zorroa.analyst.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationListener;
import org.springframework.context.event.ContextClosedEvent;
import org.springframework.stereotype.Component;

/**
 * Created by chambers on 3/13/17.
 */
@Component
public class ShutdownHook implements ApplicationListener<ContextClosedEvent> {

    @Autowired
    ProcessManagerService processManagerService;

    @Override
    public void onApplicationEvent(ContextClosedEvent contextClosedEvent) {
        processManagerService.killAllTasks();
    }

}

File: analyst/src/main/kotlin/com/zorroa/analyst/service/ShutdownHook.kt
package com.zorroa.analyst.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationListener
import org.springframework.context.event.ContextClosedEvent
import org.springframework.stereotype.Component

@Component
class ShutdownHook  @Autowired constructor(
        private val processManagerService: ProcessManagerService) : ApplicationListener<ContextClosedEvent> {

    override fun onApplicationEvent(contextClosedEvent: ContextClosedEvent) {
        processManagerService.killAllTasks()
    }

}
